### ISSUE TITLE
Add support for building static and/or dynamic library

### ISSUE THREAD

**Issue by alfredh** [baresip/re#771]
A usecase for CMake is to be able to select only static library, or dynamic library, or both.

This is used by package managers such as e.g. VCPKG

A sample solution may be found here:

https://github.com/microsoft/vcpkg/pull/30859/files#diff-0fffa68a767e7b53cac73774ef963034a32a87ebdd58ed70defc983ca564a359

**Comment by alfredh** [comment]
There is a CMake variable `BUILD_SHARED_LIBS` that can be used:

https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html

Should we use this one, or are there better options ?

**Comment by sreimers** [comment]
Yes `BUILD_SHARED_LIBS` is more the normal cmake way. With cmake migration I tried to keep the old behavior as much as possible (we build always both (static + shared)). I think we can reduce complexity and build only once at a time. I can prepare a PR for this.

**Comment by alfredh** [comment]
thanks

I think one important
usecase is Android project
from Juha

@juha-h

**Comment by juha-h** [comment]
After seeing this issue, I yesterday tested and my libbaresip-android build already builds only static re library like this:
```
libre.a: Makefile
	cd re && \
	rm -rf build && rm -rf .cache && mkdir build && cd build && \
	cmake .. \
		$(CMAKE_ANDROID_FLAGS) \
		-DCMAKE_FIND_ROOT_PATH="$(NDK_PATH);$(PWD)/openssl" \
		-DOPENSSL_VERSION_MAJOR=3 \
		-DOPENSSL_ROOT_DIR=$(PWD)/openssl && \
	cmake --build . --target re -j
```
So if something new is needed, it is for building only dynamic lib or static and dynamic libs.

**Comment by alfredh** [comment]
I dont think any changes are needed for Android, with Android I think it needs a static lib.

**Comment by alfredh** [comment]
correction:

with this patch you can select to use static or shared:

```
$ cmake -DBUILD_SHARED_LIBS=OFF .
```

**Comment by juha-h** [comment]
OK, will try after the PR has been committed.

**Comment by alfredh** [comment]
fixed in 64e5843676fb308dbe6af294623003dfd8dd8cae
